src/app/api/routes/installations.py

```diff
@@ -31,3 +32,13 @@ async def update_installation(payload: InstallationIn, installation_id: int = Pa
 @router.delete("/{installation_id}/", response_model=InstallationOut)
 async def delete_installation(installation_id: int = Path(..., gt=0)):
     return await crud.delete_entry(installations, installation_id)
+
+
+@router.get("/list_devices/{site_id}", response_model=List[int])
+async def get_all_at_given_ts_and_site(timestamp: datetime, site_id: int = Path(..., gt=0)):
+    query = installations.select(installations.c.device_id)
+    query = query.where(installations.c.site_id == site_id)
+    query = query.where(installations.c.start_ts <= timestamp)
+    query = query.where(installations.c.end_ts >= timestamp)
```

Comment:
Two things:
- rather than doing multiple where, we should use `and_` (cf. https://docs.sqlalchemy.org/en/13/core/sqlelement.html#sqlalchemy.sql.expression.and_ I'll open a PR to modify this in the rest of the codebase)
- we have to be careful, in the installation table so far, start_ts and end_ts can be None. I'm not entirely sure the query will work if that's the case?

Reply:
what is the meaning of star8ts and end_ts when it is linked to an installation?

Reply:
@florianriche it is the time range when the installation takes place (imagine retrieving a device or moving it to another site, it will fill the `end_ts` value of the current installation.

Reply:
@martin1tab regarding the earlier suggestion:
after importing `and_` from sqlalchemy, you can do :
```
    query = installations.select(installations.c.device_id).where(and_(installations.c.site_id == site_id,
                                                                       installations.c.start_ts <= timestamp,
                                                                       installations.c.end_ts >= timestamp))
```

and then simply:
```
return await database.execute(query=query)
```
(don't forget to import database)